include/xtensor/xfft.hpp

```diff
@@ -0,0 +1,317 @@
+#ifndef XFFT_HPP
+#define XFFT_HPP
+
+#include "xmath.hpp"
+#include "xcomplex.hpp"
+#include "xutils.hpp"
+#include "xaxis_slice_iterator.hpp"
+#include "xio.hpp"
+#include "xview.hpp"
+
+namespace xt 
+{
+    namespace fft
+    {
+        template<
+            typename T = double, 
+            typename E1, 
+            typename E2
+        >
+        xt::xarray<std::complex<T>> fft_convolve(E1&& xvec, E2&& yvec, std::ptrdiff_t axis = -1);
+        
+        namespace detail
+        {
+            template<typename T>
+            T pi()
+            {
+                constexpr double PI = 3.141592653589793238463;
+                return PI;
+            }
+
+            std::size_t reverse_bits(std::size_t x, int n)
+            {
+                std::size_t result = 0;
+                for (int i = 0; i < n; i++, x >>= 1)
+                    result = (result << 1) | (x & 1U);
+                return result;
+            }
+
+            template<typename E1>
+            bool is_power2(const E1& data) noexcept
+            {
+                return ((data.size() & (data.size() - 1)) == 0);
+            }
+
+            template<
+                typename T
+            >
+            auto transform_radix2(xt::xarray<std::complex<T>> data)
+            {
+                // Length variables
+                const std::size_t n = data.shape(0);
+                int levels = std::floor(std::log2(n));
+
+                // Trignometric table
+                auto i = xt::linspace<T>(0, n / 2 - 1, n / 2) * std::complex<T>(0, 1);
+                //I would like to do this but there appears to be a compiler error in CPP14
+                //Works in CPP17 and MSVC CPP14
+                //Linker error when using numeric_constants in this header
+                //auto expTable = xt::eval(xt::exp(-2 * i * ::xt::numeric_constants<T>().PI / n));
```

Comment:
```suggestion
                auto expTable = xt::eval(xt::exp(-2 * i * numeric_constants<T>::PI  / n));
```
I think that you do not need the parenthesis. Though it is probably not the solution as you tried that below it seems

Reply:
Furthermore, following naming conventions you should not use the capital. Rather `exp_table` or so

Reply:
Resolved